### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by iambenkis**
## Description
In this project, I add the [devise](https://github.com/heartcombo/devise) gem to my app and use it for the registration and login of users.

### Project requirements :1st_place_medal: 
- Remove `current_user` method in `ApplicationController`, devise will provide us with one. :heavy_check_mark: 
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started). :heavy_check_mark: 
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling. :heavy_check_mark:

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (FrederickMih) [inline app/models/user.rb:5]: - [ ] Based on the requirements of this project, this app should ask for email confirmations hence `:confirmable` should be enabled in the user model and make sure that a confirmation message is sent to a new user after registering to confirm his/her email.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @iambenkis, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 

- PR is descriptive and professional :+1:
- Great commit history :+1: 
- All Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

N/A

_Check the comments under the review._

## Optional suggestions

- Kindly remove the Microverse logo in your README.md file since you are the owner of this app and not Microverse. You can replace the logo with your own.
- Kindly complete the  **FAQ** to render your README file descriptive and professional. The license link is not working. You need to fix it.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
   has_many :posts, foreign_key: 'author_id'
```
Comment (sja-thedude) [inline app/models/user.rb]: - [ ] Kindly make sure to add confirmable as requested by the previous reviewer [here](https://github.com/iambenkis/Blog-App/pull/10/files/45942c4a66ee6ebbd6533221083c7ccf3fb9d2a6#r1049550996).

When `:confirmable` is enabled in the user model (`user.rb` file). You should see something similar to what is below:
```
devise :database_authenticatable, :registerable,
	:recoverable, :rememberable, :validatable, :confirmable
```

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @iambenkis, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job trying harder, you are almost there

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Always adding the key features of your project in the readme 👍🏿 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**